src/pie-chart/labels.tsx

```diff
@@ -78,7 +78,9 @@ export default <T extends PieChartProps.Datum>({
   containerRef,
 }: LabelsProps<T>) => {
   const containerBoundaries = useElementBoundaries(containerRef);
-
+  const shouldOptimizeLabels =
+    containerBoundaries.right - containerBoundaries.left - (dimensions.outerRadius + dimensions.innerLabelPadding) * 2 <
+    300;
```

Comment:
last little nit, leave it up to you if you want to change, this value could be assigned a descriptive variable, so anyone can easily identify or change this threshold.